modules/indexExchangeBidAdapter.js

```diff
@@ -663,11 +815,305 @@ var IndexExchangeAdapter = function IndexExchangeAdapter() {
     bidmanager.addBidResponse(firstAdUnitCode, bid);
   }
 
+  function createCygnusRequest(siteID, cygnusRequest) {
+    let cygnusUrl = (window.location.protocol === 'https:') ? url.parse(BASE_CYGNUS_VIDEO_URL_SECURE) : url.parse(BASE_CYGNUS_VIDEO_URL_INSECURE);
+    cygnusUrl.search.s = siteID;
+    cygnusUrl.search.r = encodeURIComponent(JSON.stringify(cygnusRequest));
+    let formattedCygnusUrl = url.format(cygnusUrl);
+    return formattedCygnusUrl;
+  }
+
+  /* Notify Prebid of bid responses so bids can get in the auction */
+  $$PREBID_GLOBAL$$.handleCygnusResponse = function (response) {
+    if (!response || !response.seatbid || utils.isEmpty(response.seatbid)) {
+      utils.logInfo('Cygnus returned no bids');
+
+      // signal this response is complete
+      Object.keys(bidRequests)
+        .forEach(bidId => {
+          let prebidRequest = bidRequests[bidId].prebid;
+          let bid = createBidObj(STATUS.NO_BID, prebidRequest);
+          utils.logInfo(JSON.stringify(bid));
+          bidmanager.addBidResponse(prebidRequest.placementCode, bid);
```

Comment:
Maybe I'm missing something, but I don't see anywhere where you're clearing out stale bids from `bidRequests`, yet you're iterating through all bids in `bidRequests` here.  Won't you just resubmit a bunch of old bids if multiple auctions are held?